src/policies/exponential_backoff.rs

```diff
@@ -1,28 +1,60 @@
-use crate::{RetryDecision, RetryPolicy};
-use chrono::Utc;
+use crate::{Jitter, RetryDecision, RetryPolicy};
+use chrono::{DateTime, Utc};
 use rand::distributions::uniform::{UniformFloat, UniformSampler};
-use std::{cmp, time::Duration};
+use std::{
+    cmp::{self, min},
+    time::Duration,
+};
 
-const MIN_JITTER: f64 = 0.0;
-const MAX_JITTER: f64 = 3.0;
-
-/// We are using the "decorrelated jitter" approach detailed here:
-/// <https://aws.amazon.com/blogs/architecture/exponential-backoff-and-jitter/>
+/// Exponential backoff with optional jitter.
 #[derive(Debug, Clone, Copy, PartialEq, Eq)]
+#[non_exhaustive]
 pub struct ExponentialBackoff {
     /// Maximum number of allowed retries attempts.
-    pub max_n_retries: u32,
-    /// Minimum waiting time between two retry attempts (it can end up being lower due to jittering).
+    pub max_n_retries: Option<u32>,
+    /// Minimum waiting time between two retry attempts (it can end up being lower when using full jitter).
     pub min_retry_interval: Duration,
     /// Maximum waiting time between two retry attempts.
     pub max_retry_interval: Duration,
-    /// Growing factor governing how fast the retry interval increases with respect to the number
-    /// of failed attempts. If set to 3:
-    /// - first retry: 3^0 = 1
-    /// - second retry: 3^1 = 3
-    /// - third retry: 3^2 = 9
-    /// ...
-    pub backoff_exponent: u32,
+    /// How we apply jitter to the calculated backoff intervals.
+    pub jitter: Jitter,
+}
+
+/// Exponential backoff with a maximum retry duration.
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub struct ExponentialBackoffTimed {
+    /// Maximum duration the retries can continue for, after which retries will stop.
+    max_total_retry_duration: Option<Duration>,
```

Comment:
I think I may be missing something, wouldn't we always want a set `max_total_retry_duration` if we have an `ExponentialBackoffTimed`? Else we could just use an `ExponentialBackoff`, no?

Reply:
Oh goodness, yes. Left over from an older design attempt. I'll fix that!